tap_yotpo/streams/collections.py

```diff
@@ -0,0 +1,70 @@
+"""tap-yotpo collections stream module."""
+from typing import Dict, List
+
+from singer import Transformer, get_logger, metrics, write_record
+from singer.utils import strftime, strptime_to_utc
+
+from ..helpers import ApiSpec
+from .abstracts import IncrementalStream, UrlEndpointMixin
+
+LOGGER = get_logger()
+DATE_FORMAT = "%Y-%m-%d"
+
+
+class Collections(IncrementalStream, UrlEndpointMixin):
+    """class for collections stream."""
+
+    stream = "collections"
+    tap_stream_id = "collections"
+    key_properties = ["yotpo_id"]
+    replication_key = "updated_at"
+    valid_replication_keys = ["updated_at"]
+    api_auth_version = ApiSpec.API_V3
+    config_start_key = "start_date"
```

Comment:
Instead of defining these params in all classes, we can define these params in BaseStream class and override them when need to change the value.

Reply:
hey @dbshah1212 
These properties are a part of the `BaseStream`, which is a abstract class, hence these attributes are declared as  abstract too. We have kept it to be explicitly required to define these properties on each stream class.
This prevents the catalog being generated for partially declared streams and avoids any unexpected results, and these properties are specific to `Incremental` type streams which will not be applicable for `Fulltable` type streams, also using default values with abstract classes that accommodate different types of streams might not be correct.

Reply:
Agree with your point that we should not define that in BaseStream but if we have replication_key = updated_at for most of the streams, what's the harm to mentioning that in incremental class instead of all class separately?

Reply:
The IncrementalStream does not have context to `updated_at` as it is not directly associated with any stream,
I declared it explicitly because having these details like `replication_key`, `key_properties`, `api_auth_version` on the stream class itself provides better context and readability while looking at an individual stream's implementation.